bootstrap/config/environment.go

```diff
@@ -17,61 +17,228 @@ package config
 import (
 	"fmt"
 	"os"
+	"reflect"
 	"strconv"
-
-	"github.com/edgexfoundry/go-mod-bootstrap/config"
+	"strings"
 
 	"github.com/edgexfoundry/go-mod-configuration/pkg/types"
-
 	"github.com/edgexfoundry/go-mod-core-contracts/clients/logger"
+	"github.com/pelletier/go-toml"
+
+	"github.com/edgexfoundry/go-mod-bootstrap/config"
 )
 
 const (
-	envKeyUrl             = "edgex_configuration_provider"
-	envKeyStartupDuration = "startup_duration"
-	envKeyStartupInterval = "startup_interval"
+	envKeyConfigUrl       = "EDGEX_CONFIGURATION_PROVIDER"
+	envKeyRegistryUrl     = "edgex_registry"   // TODO: Remove for v2.0.0
+	envKeyStartupDuration = "startup_duration" // TODO: Change to EDGEX_STARTUP_DURATION for v2.0.0
+	envKeyStartupInterval = "startup_interval" // TODO: Change to EDGEX_STARTUP_INTERVAL for v2.0.0
 )
```

Comment:
Could make these eg
```
v1envKeyStartupDuration = "startup_duration"
envKeyStartupDuration = "EDGEX_STARTUP_DURATION"
```
and test for both (if the new env var exists, use that, otherwise check the v1 env var)
Then the documentation for v1.2 could use the new names

Reply:
I like this. Then in v2.0 we simply remove the lowercase v1 versions.